common/cache/lru.go

```diff
@@ -62,6 +68,10 @@ type (
 		value      interface{}
 		refCount   int
 	}
+
+	hasSizeMethod interface {
+		Size() int
+	}
```

Comment:
maybe call this "TokenGetter" ?

e.g. say the cache has max token X
each item can consume Y token